PixelHunter1995/Components/SpeechLine.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace PixelHunter1995.Components
+{
+    internal class SpeechLine : IUpdateable
+    {
+        private string ActiveSpeech;
+        private TimeSpan SpeechEndTime;
+        public bool Speaking { get; private set; }
+        private bool ShouldStartNextLine;
+        private readonly Queue UpcomingLines;
```

Comment:
By the way, isn't it possible to set `Queue<string>` as the type, and then not require a cast when dequeing?